client-config/src/main/java/com/palantir/conjure/java/client/config/ClientConfigurations.java

```diff
@@ -163,6 +168,19 @@ private static ClientConfiguration of(
                 .build();
     }
 
+    private static ImmutableSet<String> parseNoProxyEnvironmentVar() {
+        String env = System.getenv(ENV_NO_PROXY);
+        if (env == null || env.isEmpty()) {
+            log.debug(ENV_NO_PROXY + " environment variable not set.");
+            return ImmutableSet.of();
+        }
+        return Arrays.stream(env.split(","))
+                .map(String::trim)
+                .filter(s -> !s.isEmpty())
+                .map(String::toLowerCase)
+                .collect(ImmutableSet.toImmutableSet());
```

Comment:
As a matter of testing, it might be useful to test IP addresses in the no_proxy config too (not sure if this is before or after DNS-resolution, so this might be either matching on the IP-address itself, or the DNS-resolved IP address from the hostname?)

If we don't want to support IP address at all, maybe a log line indicating we found IP addresses in the no_proxy config and these aren't supported?